common/src/main/java/org/openmrs/analytics/HttpUtil.java

```diff
@@ -30,10 +32,10 @@
 	
 	private static final Logger log = LoggerFactory.getLogger(FhirStoreUtil.class);
 	
-	public String executeRequest(HttpUriRequest request) {
+	public String executeRequest(HttpUriRequest request) throws IOException {
+		CloseableHttpClient httpClient = HttpClients.createDefault();
```

Comment:
Following up on what I suggested above, it seems `AutoRetryHttpClient` of the Apache HTTP lib, may already implement what I am suggesting; WDYT?

Reply:
[AutoRetryHttpClient](https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/impl/client/AutoRetryHttpClient.html) is deprecated , let me use the recomeneded HttpClientBuilder instead

Reply:
I see.